lib/src/handle_access_token_response.dart

```diff
@@ -117,8 +149,10 @@ void _handleErrorResponse(http.Response response, Uri tokenEndpoint) {
       ? null
       : new MediaType.parse(contentTypeString);
 
-  validate(contentType != null && contentType.mimeType == "application/json",
-      'content-type was "$contentType", expected "application/json"');
+  validate(contentType != null &&
+      (contentType.mimeType == "application/json" || contentType.mimeType == "text/plain" ||
+       contentType.mimeType == "application/x-www-form-urlencoded"),
+      'content-type was "$contentType", expected "application/json", "application/x-www-form-urlencoded", or "text/plain"');
```

Comment:
This logic doesn't allow for extensible parameter-parsing. This check should be in the default `getParameters()`.